curp/src/error.rs

```diff
@@ -65,7 +65,7 @@ pub enum ServerError {
 }
 
 /// The error met during propose phase
-#[derive(Error, Debug, Clone, Serialize, Deserialize)]
+#[derive(Error, Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
```

Comment:
Why should we derive the `PartialEq` and the `Eq` for these error types?  They are enum type, so using `matches!` will be more reasonable.

Reply:
done